### DISCUSSION TITLE
DNSSEC-free resolvers only

### DISCUSSION THREAD

**Discussion by flawrist** [DNSCrypt/dnscrypt-proxy#2571] [Q&A]
Tonight a lot of Russian users faced the inability to open `.ru` websites.

It was [covered](https://t.me/mintsifry/2115) by the Ministry of Digital Development and [local media](https://www.rt.com/russia/591566-major-internet-outage-reported-russia/), I myself faced it. The issue turned out to be related to the "global DNSSEC infrastructure". What exactly this means remains unclear. Perhaps this is someone's mistake, perhaps malicious sabotage due to the ongoing cold war, but this is definitely not a power outage.

A workaround during these hours, which helped some of us, was to switch to the ISP's DNS resolver without DNSSEC. Now, how do I do the same with DNSCrypt if its option `require_dnssec = false` returns both kind of resolvers (with and without DNSSEC)?

**Comment by lifenjoiner** [comment]
> how do I do the same with DNSCrypt if its option require_dnssec = false returns both kind of resolvers (with and without DNSSEC)?

Like DoHTTP**S**, DNSSEC, DNSCrypt is also an encryption protocol.

```
DNSCrypt/HTTPS {
    DNSSEC {
        Plain DNS {}
    }
}
```
So, DNSSEC is an inner layer here. I don't think requiring DNSSEC or not could be known from the outside.
DNSSEC takes effect during the upstream server communicates with its upstream servers, when this layer becomes the outer layer.

**Reply by flawrist** [reply]
I found a tool to diagnose the features of DNS resolvers:
https://cmdns.dev.dns-oarc.net/

After diagnosing a few resolvers, I found that DNSSEC is not supported
at least by `libredns` and `yandex`. I will update if it happens again.

![2024-0131-1453 checkmydns](https://github.com/DNSCrypt/dnscrypt-proxy/assets/373718/9e1b86b9-1022-4ceb-91b8-f6da81941ce7)
